packages/wdio-browserstack-service/tests/launcher.test.ts

```diff
@@ -12,9 +12,31 @@ import type { Capabilities, Options } from '@wdio/types'
 import BrowserstackLauncher from '../src/launcher.js'
 import type { BrowserstackConfig } from '../src/types.js'
 import * as utils from '../src/util.js'
+import * as bstackLogger from '../src/bstackLogger.js'
 
 vi.mock('@wdio/logger', () => import(path.join(process.cwd(), '__mocks__', '@wdio/logger')))
 vi.mock('browserstack-local')
+vi.mock('fs/promises', () => ({
+    default: {
+        createReadStream: vi.fn().mockReturnValue({ pipe: vi.fn() }),
+        createWriteStream: vi.fn().mockReturnValue({ pipe: vi.fn() }),
+        stat: vi.fn().mockReturnValue(Promise.resolve({ size: 123 })),
+    }
+}))
+
+vi.mock('form-data-node', () => vi.fn().mockReturnValue({
+    append: vi.fn()
+}))
+
+vi.spyOn(utils, 'uploadLogs').mockImplementation((_user, _key, _uuid) => new Promise((resolve) => {
```

Comment:
Can we not mock this globally as this is getting used at only one fuction? This will unncessarily interfere with tests that may require it